tests/test_sockets.py

```diff
@@ -1849,3 +1850,28 @@ async def test_connect_tcp_getaddrinfo_context() -> None:
             pass
 
     assert exc_info.value.__context__ is None
+
+
+async def test_wait_socket_readable(anyio_backend_name: str) -> None:
+    if anyio_backend_name == "trio" and platform.system() == "Windows":
+        pytest.skip("Internal error in Trio")
+
+    def client(port: int) -> None:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.connect(("127.0.0.1", port))
+            sock.sendall(b"Hello, world")
+
+    with move_on_after(0.1):
```

Comment:
Why a timeout this strict? It's bound to cause flakiness on slow/busy systems.
Also, wouldn't `fail_after` eliminate the need to set a flag?
```suggestion
    with fail_after(5):
```
On another note, I think this should be on the same line as `with conn:`, given that it's the only place where the timeout can work.